### PR TITLE
msvc_command and vcvars improvements

### PR DESCRIPTION
**Pull request by memsharded**
Address https://github.com/conan-io/conan/issues/1130

### CODE REVIEW

File: conans/tools.py [hunk 1]
```diff
@@ -121,7 +121,18 @@ def environment_append(env_vars):
         os.environ.update(old_env)
 
 
-def build_sln_command(settings, sln_path, targets=None, upgrade_project=True):
+def msvc_command(settings, sln_path, targets=None, upgrade_project=True, build_type=None,
+                 arch=None):
+    """ Do both: set the environment variables and call the .sln build
+    """
+    vcvars = vcvars_command(settings)
```
Comment (lasote) [inline conans/tools.py]: And... where are you controlling that vcvars doesn't return empty?

File: conans/tools.py [hunk 2]
```diff
@@ -121,7 +121,18 @@ def environment_append(env_vars):
         os.environ.update(old_env)
 
 
-def build_sln_command(settings, sln_path, targets=None, upgrade_project=True):
+def msvc_command(settings, sln_path, targets=None, upgrade_project=True, build_type=None,
```
Comment (lasote) [inline conans/tools.py]: Hard to know what it does reading the name, specially having the build_sln_command...

File: conans/tools.py (hunk 1 above)
Comment (memsharded) [inline conans/tools.py]: Not necessary, it has been replaced by an "echo" command, not empty

File: conans/tools.py (hunk 2 above)
Comment (memsharded) [inline conans/tools.py]: Suggestions welcome :)

File: conans/tools.py [hunk 3]
```diff
@@ -146,7 +164,7 @@ def vcvars_command(settings):
     param = "x86" if settings.arch == "x86" else "amd64"
     existing_version = os.environ.get("VisualStudioVersion")
     if existing_version:
-        command = ""
+        command = "echo Conan:vcvars already set"
```
Comment (lasote) [inline conans/tools.py]: Could you add a test that checks that output to ensure that it's running ok when vcvars is already set? I know that probably it's already tested, but just for ensure it.

File: conans/tools.py (hunk 1 above)
Comment (lasote) [inline conans/tools.py]: I would prefer `build_sln_with_vcvars_command` or similar. I know that it's long but better long but meaninful.

File: conans/tools.py (hunk 2 above)
Comment (lasote) [inline conans/tools.py]: I would prefer build_sln_with_vcvars_command or similar. I know that it's long but better long than meanless.

**Pull request merged**